### ISSUE TITLE
Model edition does not take effects

### ISSUE THREAD

**Issue by JX-Zhang98** [zjunlp/EasyEdit#163]
I tried to edit LLaMa-7b with ROME and tell it some fake fact. But it doesn't work...

Here is the config :
```
alg_name: "ROME"
model_name: "/models/Llama-2-7b-chat-hf"
stats_dir: "./data/stats"
device: 0
layers: [5]
fact_token: "subject_last"
v_num_grad_steps: 25
v_lr: 5e-1
v_loss_layer: 31
v_weight_decay: 1e-3
clamp_norm_factor: 4
kl_factor: 0.0625
mom2_adjustment: false
context_template_length_params: [[5, 10], [10, 10]]
rewrite_module_tmp: "model.layers.{}.mlp.down_proj"
layer_module_tmp: "model.layers.{}"
mlp_module_tmp: "model.layers.{}.mlp"
attn_module_tmp: "model.layers.{}.self_attn"
ln_f_module: "model.norm"
lm_head_module: "lm_head"
mom2_dataset: "wikipedia"
mom2_n_samples: 100000
mom2_dtype: "float32"
model_parallel: true
```
And my edit script is：
```
def test_ROME_LlaMA():
    # prompts = ['Where is the capital of China?']
    # ground_truth = ['Beijing']
    # target_new = ['New York']
    # subject = ['China']
    prompts = ['中国的首都是哪里？']
    ground_truth = ['北京']
    target_new = ['纽约']
    subject = ['中国']
    hparams = ROMEHyperParams.from_hparams('./hparams/ROME/llama-7b.yaml')
    editor = BaseEditor.from_hparams(hparams)

    input_token = editor.tok(prompts, return_tensors='pt', padding=True)
    pre_edit_output = editor.model.generate(
        input_ids = input_token['input_ids'].to('cuda'),
        attention_mask=input_token['attention_mask'].to('cuda'),
        max_length = 32
    )
    print(utils.Color.green(editor.tok.decode(pre_edit_output[0])))


    metrics, edited_model, _ = editor.edit(
        prompts=prompts,
        ground_truth=ground_truth,
        target_new=target_new,
        subject=subject,
        keep_original_weight=True
    )

    post_edit_output = edited_model.generate(
        input_ids = input_token['input_ids'].to('cuda'),
        attention_mask=input_token['attention_mask'].to('cuda'),
        max_length = 32
    )
    print(utils.Color.yellow(editor.tok.decode(post_edit_output[0])))
    return metrics, edited_model, editor.tok
```
I tried to tell the model that the capital of china is New York, but it still say Beijing after model editing.
The process of running the script looks all right, but the output of the edited model is still the ground truth, not the set target_new.
The output of the scripts is ：
```
2024-01-25 11:03:04,131 - easyeditor.editors.editor - INFO - Instantiating model
01/25/2024 11:03:04 - INFO - easyeditor.editors.editor -   Instantiating model
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:13<00:00,  6.65s/it]
<s> 中国的首都是哪里？

China's capital city is Beijing.</s>
Executing ROME algorithm for the update: [中国的首都是哪里？] -> [ ICSL]
Cached context templates ['{}', 'The 4 Best. {}', 'The first step in. {}', 'Therefore, the company. {}', 'Therefore, I will. {}', 'Because the 1. {}', 'Because of the high. {}', 'I was born and. {}', 'I have to give. {}', 'You are here:. {}', 'You are here:. {}', 'The Southeast Asian city-state of. {}', 'The 2023 NFL Draft. {}', 'Therefore, the first step in creating an effective. {}', 'Therefore, it is essential to understand the factors. {}', 'Because of its high concentration of antiox. {}', 'Because of his unique style and approach to comedy. {}', "I'm not going to sugarcoat. {}", "I'm not sure if you've. {}", 'You are here: Home / Blog /. {}', 'You are here: Home / Blog /. {}']
Computing left vector (u)...
Selected u projection object 中国
Left vector shape: torch.Size([11008])
Computing right vector (v)
Lookup index found: 3 | Sentence: 中国的首都是哪里？ ICS | Token: 国
Rewrite layer is 5
Tying optimization objective to 31
Recording initial value of v*
loss 7.747 = 7.747 + 0.0 + 0.0 avg prob of [ ICSL] 0.0005219830782152712
loss 6.817 = 6.711 + 0.104 + 0.001 avg prob of [ ICSL] 0.0013496471801772714
loss 5.229 = 4.871 + 0.357 + 0.001 avg prob of [ ICSL] 0.008436986245214939
loss 3.307 = 2.916 + 0.39 + 0.001 avg prob of [ ICSL] 0.05641317367553711
loss 2.409 = 2.137 + 0.271 + 0.001 avg prob of [ ICSL] 0.12676706910133362
loss 1.136 = 0.949 + 0.185 + 0.001 avg prob of [ ICSL] 0.3964758813381195
loss 0.301 = 0.172 + 0.128 + 0.001 avg prob of [ ICSL] 0.8429136276245117
loss 0.159 = 0.048 + 0.11 + 0.001 avg prob of [ ICSL] 0.9540724158287048
loss 0.109 = 0.005 + 0.103 + 0.001 avg prob of [ ICSL] 0.9945564270019531
loss 0.101 = 0.004 + 0.096 + 0.001 avg prob of [ ICSL] 0.9959218502044678
loss 0.092 = 0.003 + 0.088 + 0.001 avg prob of [ ICSL] 0.9967846870422363
loss 0.084 = 0.002 + 0.081 + 0.001 avg prob of [ ICSL] 0.9976708889007568
loss 0.078 = 0.002 + 0.076 + 0.001 avg prob of [ ICSL] 0.998325526714325
loss 0.073 = 0.001 + 0.071 + 0.001 avg prob of [ ICSL] 0.9986926913261414
loss 0.067 = 0.001 + 0.065 + 0.001 avg prob of [ ICSL] 0.9988481402397156
loss 0.061 = 0.001 + 0.059 + 0.001 avg prob of [ ICSL] 0.9988601803779602
loss 0.055 = 0.001 + 0.053 + 0.001 avg prob of [ ICSL] 0.9987925887107849
loss 0.048 = 0.001 + 0.045 + 0.001 avg prob of [ ICSL] 0.9987112879753113
Delta norm: 14.623559951782227
Change in target norm: 3.6558899879455566 to 15.08608341217041 => 11.430192947387695
Division Factor: 2.8578109741210938
Right vector norm: 5.117049217224121
Right vector shape: torch.Size([4096])
Deltas successfully computed for ['model.layers.5.mlp.down_proj.weight']
New weights successfully inserted into ['model.layers.5.mlp.down_proj.weight']
2024-01-25 11:03:51,830 - easyeditor.editors.editor - INFO - Execution 0 editing took 25.046738862991333
01/25/2024 11:03:51 - INFO - easyeditor.editors.editor -   Execution 0 editing took 25.046738862991333
2024-01-25 11:03:51,938 - easyeditor.editors.editor - INFO - Evaluation took 0.1074666976928711
01/25/2024 11:03:51 - INFO - easyeditor.editors.editor -   Evaluation took 0.1074666976928711
2024-01-25 11:03:51,938 - easyeditor.editors.editor - INFO - 0 editing: 中国的首都是哪里？ -> ICSL  
 {'pre': {'rewrite_acc': [0.0], 'portability': {}}, 'case_id': 0, 'requested_rewrite': {'prompt': '中国的首都是哪里？', 'target_new': 'ICSL', 'ground_truth': '北京', 'portability': {}, 'locality': {}, 'subject': '中国'}, 'time': 25.046738862991333, 'post': {'rewrite_acc': [1.0], 'locality': {}, 'portability': {}}}
01/25/2024 11:03:51 - INFO - easyeditor.editors.editor -   0 editing: 中国的首都是哪里？ -> ICSL  
 {'pre': {'rewrite_acc': [0.0], 'portability': {}}, 'case_id': 0, 'requested_rewrite': {'prompt': '中国的首都是哪里？', 'target_new': 'ICSL', 'ground_truth': '北京', 'portability': {}, 'locality': {}, 'subject': '中国'}, 'time': 25.046738862991333, 'post': {'rewrite_acc': [1.0], 'locality': {}, 'portability': {}}}
<s> 中国的首都是哪里？

China's capital city is Beijing.</s>
save model? [y/n]n
```
Is there anything wrong with my operation? What should I do to make the model edit correctly?
Thank you

**Comment by JX-Zhang98** [comment]
In addition，even if I provide the prompt in English, it still cannot be modified effectively.
Though the loss is low enough and the 'rewrite_acc' is  [1.0]...

**Comment by littlefive5** [comment]
I will reimplement your code and will let you know if I find out the problem.

**Comment by littlefive5** [comment]
Can you set keep_original_weight=Fasle and Try again?

**Comment by JX-Zhang98** [comment]
Thank you，I will try it later



---Original---
From: "Yunzhi ***@***.***&gt;
Date: Thu, Jan 25, 2024 12:12 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [zjunlp/EasyEdit] Model edition does not take effects (Issue#163)




 
Can you set keep_original_weight=Fasle and Try again?

**Comment by zxlzr** [comment]
Hi, do you have any other questions?

**Comment by JX-Zhang98** [comment]
It seems works.
I will do more exploring, thanks for your help

**Comment by fateme-hshm96** [comment]
Hello. Thanks for this great repo. 
I'm facing the same issue. I set keep_original_weight to false.
I tried test_ROME_Mistral and test_MEMIT_Mistral. The outputs are exactly the same before and after the edit.

**Comment by fateme-hshm96** [comment]
An update on this: It seems like the original weights are restored after the edit, regardless of `keep_original_weight` variable.
An example in MEMIT code:
https://github.com/zjunlp/EasyEdit/blob/main/easyeditor/models/memit/memit_main.py#L235
